### ISSUE TITLE
[A11y] Identify Flutter semantics performance issues.

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#150234]
### Steps to reproduce

tldr: the goal is to find out how to improve the performance when turning on semantics and scope out work to improve that.

When turning on semantics in Flutter, the app may suffer performance issue due to semantics tree compilation.

We used to only turn on semantics conditionally when needed, but we started to get use cases where we want to turn on by default. For example

1. web platform can't tell whether a assistive technology is running
2. web SEO uses semantics tree to index
3. android app crawl also use accessibilitynodes to crawl the app.

There the things we should do now is to figure out
1. Whether the performance is really bad enough to prevent us from doing that
4. if the answer to 1 is yes, What causes the performance issue
5. if the answer to 1 is yes, What needs to be done to make it run faster.

### related issues

Scrolling jank https://github.com/flutter/flutter/issues/61193

**Comment by cedvdb** [comment]
Could you explain what do you mean by your point 2 ?

How could this be related to SEO ? It is my understanding that SEO bots don't run javascript nor wasm, so they basically just see an empty index.html

**Comment by chunhtai** [comment]
SEO crawls dom element tree. Since we create a dom tree that repesent flutter app when semantics is turned on for semantics purposes, we think it makes more sense that we can use this dom tree for SEO crawl as well.

The problem is then the semantic dom tree may not always be there when SEO happens, there are workarounds for that, but it involves some hack

**Comment by cedvdb** [comment]
@chunhtai SEO crawlers don't execute javascript though, do they ?

 > Since we create a dom tree

By "we" I assume you mean flutter, which on web is javascript ( which is not executed by crawlers).

**Comment by chunhtai** [comment]
some more discoveries
![image](https://github.com/flutter/flutter/assets/47866232/bb3f4035-3318-4294-acad-67f941bd9666)
semantics update take about 2.6ms in a 6ms frame when scrolling a static content. so almost 50%

**Comment by chunhtai** [comment]
Change is in https://github.com/flutter/flutter/pull/161195, this should have cut the performance significantly. latest number is 1.6ms ->0.3ms in a 4ms frame
